### DISCUSSION TITLE
entity ruler negate operator behavior

### DISCUSSION THREAD

**Discussion by codybakken** [explosion/spaCy#8957] [Help: Coding & Implementations] [answered]
I'm trying to understand why this pattern works. I expect it to capture 'application' but not 'application fee'. I used the matcher demo and it worked fine. Does the '!' operator not work with the entity_ruler? I want to capture 'application' but not 'application fee.'

```
nlp = spacy.load('en_core_web_lg')
ruler = nlp.add_pipe("entity_ruler")
patterns = [{'label':'APPLICATION', 'pattern': [{'LOWER': 'application'},{'LOWER': 'fee','OP':'!'}]}]
ruler.add_patterns(patterns)

text = 'application' #returns nothing
doc = nlp(text)

for ent in doc.ents:
    print(ent.label_,ent.text,ent.start_char,ent.end_char)
```

**Comment by michelleful** [comment, marked as answer]
The fact that there is another token defined afterwards in the pattern, i.e. `{'LOWER': 'fee', 'OP': '!'}`, means that it requires a following token, therefore it will not match the situation where "application" is the last token. If this situation were at the start, you could use the IS_SENT_START attribute, but I'm not sure what is advised in an end-of-sentence situation. You may need to define your own custom _.IS_SENT_END attribute.
